src/app/components/NavigationBar/NavigationBar.blocks.tsx

```diff
@@ -0,0 +1,159 @@
+import { images } from "app/assets/images";
+import { Button } from "app/components/Button";
+import { Icon } from "app/components/Icon";
+import { NotificationsDropdown } from "app/components/Notifications";
+import { Tooltip } from "app/components/Tooltip";
+import { useConfiguration } from "app/contexts";
+import { useScroll } from "app/hooks";
+import { ReceiveFunds } from "domains/wallet/components/ReceiveFunds";
+import { SearchWallet } from "domains/wallet/components/SearchWallet";
+import { SelectedWallet } from "domains/wallet/components/SearchWallet/SearchWallet.models";
+import React, { useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useHistory } from "react-router-dom";
+import { openExternal } from "utils/electron-utils";
+
+import { BackButton } from "./components/BackButton";
+import { Balance } from "./components/Balance";
+import { UserInfo } from "./components/UserInfo";
+import { NavigationBarProps } from "./NavigationBar.contract";
+import { getUserInitials, getWallets, renderMenuLinks } from "./NavigationBar.helpers";
+import { LogoContainer, NavigationButtonWrapper, NavWrapper } from "./NavigationBar.styles";
+
+const { ARKLogo } = images.common;
+
+const renderBackButton = ({
+	isBackDisabled,
+	variant,
+}: NavigationBarProps) => {
+	if (variant !== "full") {
+		return;
+	}
+
+	return <BackButton className="flex w-12" disabled={isBackDisabled} />;
+};
+
+const renderMenu = ({
```

Comment:
This function is still pretty large and should probably be extracted into a separate component to work more like `renderBackButton` where a one-liner is returned for a component instead of building it inside the function.